lib/jekyll-admin/server/theme.rb

```diff
@@ -0,0 +1,117 @@
+module JekyllAdmin
+  class Server < Sinatra::Base
+    namespace "/theme" do
+      get do
+        ensure_theme
+        json(
+          {
+            :name        => theme.name,
+            :version     => theme.version,
+            :authors     => theme.authors.join(", "),
+            :license     => theme.license,
+            :path        => at_root,
+            :directories => entries.map(&:to_api),
+          }
+        )
+      end
+
+      get "/*?/?:path.:ext" do
+        ensure_theme
+        json(
+          { :name => filename,
+            :path => file_path, }.merge!(contents).merge!(url_fields)
+        )
+      end
+
+      get "/?*" do
+        ensure_theme && ensure_directory
+        json({
+          :name    => splats.first.split("/").last,
+          :path    => at_root(splats.first),
+          :entries => entries.map(&:to_api).concat(subdir_entries),
+        })
+      end
+
+      private
+
+      def ensure_theme
+        render_404 unless site.theme
+      end
+
+      def at_root(dir = "")
+        site.in_theme_dir(dir)
```

Comment:
Why do we need a helper here? If we do, can we call it `in_theme_dir` or something similar for consistency (rather than introduce a new naming convention)?